### DISCUSSION TITLE
How to use `createTestingPinia()` when pinia it is used outside a component

### DISCUSSION THREAD

**Discussion by yakobe** [vuejs/pinia#975] [Help and Questions]
Imagine i have some code (eg a composable) that uses a store. I would like to test the code with jest and mock the store with `createTestingPinia()`. However, the actions of the store are not mocked and still call api's etc.

I imagine this is because my code does not know that it should be using the pinia created by `createTestingPinia()` and maybe creates it's own instance of pinia? Is it possible to somehow so that i can spy on an action to ensure it is called? 

Many thanks ☺️

**Comment by posva** [comment]
You need to call `setActivePinia()` with the testing pinia for it to be used (https://pinia.vuejs.org/cookbook/testing.html#unit-testing-a-store)

**Reply by yakobe** [reply]
Thanks for getting back to me @posva 👍 

Do you mean something like this in the jest test: 
```js
  it('calls `myStoreAction` on the Store', async () => {
	// Given
    const pinia = createTestingPinia()
    setActivePinia(pinia)
    const myStore = useMyStore()
    const myStoreAction = jest.spyOn(myStore, 'myStoreAction')

    // When
    await myFunctionUnderTest('some param')

    // Then
    expect(myStoreAction).toHaveBeenCalledTimes(1)
  })
```

Or have i misunderstood? With the above code the `myStoreAction` is not mocked and runs the real code with calls to APIs etc. I hoped with `createTestingPinia()` it would be mocked.

**Reply by posva** [reply]
Yes. You can pass an options object to createTestingPinia. Check the available options, there is one to stub actions. I think there is also an example in docs in the link I sent you

**Reply by yakobe** [reply]
Hi @posva 

After some digging around, the option `fakeApp: true` seems to do what i need. I'm guessing it creates a fake app so we always get the same instance. Does that sound about right? Seems to work 😄.
